python_app/Test_SoundGeneratorVAE_V2/autoencoder.py

```diff
@@ -0,0 +1,256 @@
+        optimizer = Adam(learning_rate=learning_rate)
+        self.model.compile(optimizer=optimizer,
+                           loss=self._calculate_combined_loss,
+                           metrics=[self._calculate_reconstruction_loss,
+                                    self._calculate_kl_loss])
+
+    def train(self, x_train, batch_size, num_epochs):
+        self.model.fit(x_train,
+                       x_train,
+                       batch_size=batch_size,
+                       epochs=num_epochs,
+                       shuffle=True)
+
+    def save(self, save_folder="."):
+        self._create_folder_if_it_doesnt_exist(save_folder)
+        self._save_parameters(save_folder)
+        self._save_weights(save_folder)
+
+    def load_weights(self, weights_path):
+        self.model.load_weights(weights_path)
+
+    def reconstruct(self, images):
+        latent_representations = self.encoder.predict(images)
+        reconstructed_images = self.decoder.predict(latent_representations)
+        return reconstructed_images, latent_representations
+
+    @classmethod
+    def load(cls, save_folder="."):
+        parameters_path = os.path.join(save_folder, "parameters.pkl")
+        with open(parameters_path, "rb") as f:
+            parameters = pickle.load(f)
+        autoencoder = VAE(*parameters)
+        weights_path = os.path.join(save_folder, "weights.h5")
+        autoencoder.load_weights(weights_path)
+        return autoencoder
+
+    def _calculate_combined_loss(self, y_target, y_predicted):
+        reconstruction_loss = self._calculate_reconstruction_loss(y_target, y_predicted)
+        kl_loss = self._calculate_kl_loss(y_target, y_predicted)
+        combined_loss = self.reconstruction_loss_weight * reconstruction_loss\
+                                                         + kl_loss
+        return combined_loss
+
+    def _calculate_reconstruction_loss(self, y_target, y_predicted):
+        error = y_target - y_predicted
+        reconstruction_loss = K.mean(K.square(error), axis=[1, 2, 3])
+        return reconstruction_loss
+
+    def _calculate_kl_loss(self, y_target, y_predicted):
+        kl_loss = -0.5 * K.sum(1 + self.log_variance - K.square(self.mu) -
+                               K.exp(self.log_variance), axis=1)
+        return kl_loss
+
+    def _create_folder_if_it_doesnt_exist(self, folder):
+        if not os.path.exists(folder):
+            os.makedirs(folder)
+
+    def _save_parameters(self, save_folder):
+        parameters = [
+            self.input_shape,
+            self.conv_filters,
+            self.conv_kernels,
+            self.conv_strides,
+            self.latent_space_dim
+        ]
+        save_path = os.path.join(save_folder, "parameters.pkl")
+        with open(save_path, "wb") as f:
+            pickle.dump(parameters, f)
+
+    def _save_weights(self, save_folder):
+        save_path = os.path.join(save_folder, "weights.h5")
+        self.model.save_weights(save_path)
```

Comment:
here are the parts, that for some reason don´t work. They donb´t create these files

Reply:
hi, can you provide some error message? Maybe the reason lies in the initialisation of the model from line 244. The tensor dimensions (28,28,1) seems not to be our spectrograms (too small).

Reply:
So in the autoencoder.py script is no error :D that´s a problem. The dimensions here doesn´t matter also, because the script train.py is transfering the correct size then at the end. I tried to change the input size there also but still no luck. Here is my error code from the train.py:
AttributeError: 'method' object has no attribute '_from_serialized'